### DISCUSSION TITLE
Using Andes in a closed loop environment

### DISCUSSION THREAD

**Discussion by vaahedi** [CURENT/andes#217] [Ideas]
Hello everyone

I am interested to use Andes as a simulator for PV and battery interacting with another program exchanging telemetry and control. Has anyone used Andes in a closed loop environment and if so what was the performance like

**Comment by cuihantao** [comment]
Hi @vaahedi ,

Our recent work uses ANDES for Transmission & Distribution co-simulation. See https://arxiv.org/abs/2101.05894.

The performance of ANDES would not change. The time to simulate may increase if the external program constantly alters the operating point. I would recommend writing a mock program for benchmarking if interested.

**Reply by vaahedi** [reply]
Thank you, how often where you interrogating the simulator. How fast were you getting telemetries and how fast were you deploying controls ?

**Reply by cuihantao** [reply]
I believe data exchange happens every one second in the setup. @xfangnrel can provide more details.

**Reply by XinFang88** [reply]
The exchange frequency will be decided by your purpose of the simulation. ANDES can exchange data with other programs/simulators with a defined time step.

**Reply by vaahedi** [reply]
Thank you; so let us say, I have a program that wants to control some assets every one minute and collect telemetry every 2 seconds. If we only have a a few asset, can the simulation program run very close to real time meaning that we see the behavior of the assets as if they are physical devices

**Reply by cuihantao** [reply]
I believe so but encourage you to test it out.

The IEEE 14 bus system takes 2 seconds wall-clock time to run 20 seconds of simulation. The WECC 181-bus system might take 18 seconds for 20 seconds of simulation, depending on the dynamic models you have.

You might want to check the `--qrt` flag for ``andes run -r tds`` to enable a quasi-real-time stepping.

**Reply by vaahedi** [reply]
many thanks, this information should be sufficient for my purpose now
